service/src/main/java/uk/gov/hmcts/reform/fpl/jobs/UndeliveredEmailsFinder.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.reform.fpl.jobs;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.joda.time.DateTime;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.ApplicationEventPublisher;
+import uk.gov.hmcts.reform.fpl.events.UndeliveredEmailsFound;
+import uk.gov.hmcts.reform.fpl.exceptions.JobException;
+import uk.gov.hmcts.reform.fpl.model.UndeliveredEmail;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationList;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+@ConditionalOnProperty(value = "scheduler.enabled", havingValue = "true")
+public class UndeliveredEmailsFinder implements Job {
+
+    private static final int REPORT_PERIOD_IN_DAYS = 1;
+
+    private final NotificationClient notifications;
+    private final ApplicationEventPublisher applicationEventPublisher;
+
+    @Override
+    public void execute(JobExecutionContext jobExecutionContext) {
+        final String jobName = jobExecutionContext.getJobDetail().getKey().getName();
+
+        log.info("Job '{}' started", jobName);
+
+        try {
+            DateTime reportFrom = DateTime.now().minusDays(REPORT_PERIOD_IN_DAYS);
```

Comment:
maybe more consisten plugging the clock maybe?

Reply:
it is joda time, enforced by gov notify api

Reply:
An easy way is to 1) get the java time in the main class and convert it to joda, or 2) use @before DateTimeUtils.setCurrentMillisFixed(long)/ and @after DateTimeUtils.setCurrentMillisSystem() in the test, to use without having to to through conversion. In this way the test is not time dependent of the current clock.